### PR TITLE
v1.8 backports 2020-06-11

### PR DESCRIPTION
**Pull request by aanm**
* #11980 -- loader: Fix tunneling when device is set without NodePort (@pchaigno)
 * #11947 -- Fix EndpointSlice support (@aanm)
 * #11974 -- cilium: fixes transparent encryption (@jrfastab)
 * #11982 -- daemon: Remove checkHostFirewallWithEgressLB() (@pchaigno)
 * #11989 -- bpf: enable hairpin optimizations to avoid fib lookup also for tc (@borkmann)
 * #11882 -- doc: Update the Hubble section of getting started guides (@michi-covalent)
 * #11765 -- test: Add debugging info for potential flakes in NodePort services tests with L4 and L7 (@jrajahalme)
 * #11987 -- vagrant: Fix make in net-next dev. VM (@pchaigno)
 * #11996 -- docs: Add Hubble metrics reference (@gandro)
 * #11007 -- cilium: fix up all --help sections properly (@soumynathan)
 * #11757 -- Implement values for hubble-relay to properly control sub chart values (@seanmwinn)
 * #11995 -- test: Wait for POD policy revision increment in all cases. (@jrajahalme)
 * #11973 -- Envoy: Update to Envoy 1.13.2 (@jrajahalme)
 * #11998 -- cilium, test: Only run sockops tests on 4.19 and bpf-next kernels (@jrfastab)
 * #11993 -- test: Add retries to curl command (@christarazi)
 * #11891 -- etcd: propagate Context from higher-level calls (@tklauser)
 * #12008 -- daemon, option: remove deprecated flannel-manage-existing-containers flag (@tklauser)
 * #11571 -- azure: support standalone (non VMSS) instances (@bpineau)
 * #12020 -- doc: Use --reuse-values option for helm upgrade (@michi-covalent)
 * #12016 -- envoy: Include detail in NACK warning (@jrajahalme)
 * #12007 -- test: Gather combined Cilium logs as last resort (@nebril)
 * #12023 -- helm: Use port 80 for service/hubble-ui (@gandro)
 * #11894 -- daemon: Fix detection of BPF/XDP NodePort, BPF masq and host-fw devices (@brb)
 * #12006 -- iptables: carry on and log on failure to set up transient rules (@qmonnet)
 * #12030 -- install: Fix up version/pullPolicy for multiple values files (@joestringer )

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 11980 11947 11974 11982 11989 11882 11765 11987 11996 11007 11757 11995 11973 11998 11993 11891 12008 11571 12020 12016 12007 12023 11894 12006 12030; do contrib/backporting/set-labels.py $pr done 1.8; done
```

@genbit this PR didn't apply successfully, is the 1.8 branch already with such changes?

 * #11803 -- docs: re-design cilium docs theme (@genbit)

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -64,6 +64,9 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# Can be removed once PR #11528 with the golangci-lint replacement is merged.
```
Comment (pchaigno) [inline Vagrantfile:67]: Nit: Since we don't plan to backport #11528 to v1.8, I think we should drop this comment.

**Review by pchaigno — APPROVED** [review]
:+1: for my changes!

**Review by tklauser — APPROVED** [review]
LGTM for my changes

**Review by christarazi — APPROVED** [review]

**Comment by joestringer** [comment]
This will fix the hubble-relay version issue:
https://github.com/cilium/cilium/pull/12030

File: Vagrantfile (hunk 1 above)
Comment (aanm) [reply Vagrantfile:67]: Let's keep it so we don't lose context.

**Comment by aanm** [comment]
test-backport-1.8

**Comment by genbit** [comment]
> @genbit this PR didn't apply successfully, is the 1.8 branch already with such changes?
@aanm was it due to the conflicts in Documentation/requirements.txt ?

**Comment by genbit** [comment]
@aanm  I've created PR with two docs changes https://github.com/cilium/cilium/pull/12032

**Comment by joestringer** [comment]
All failures are because of the hubble-relay image pull issue which is likely because of a lack of `v1.8.0-rc2` image tag on dockerhub. I will tag that now.

**Comment by joestringer** [comment]
Given that all tests are currently passing, I will push the fix for the hubble issue and focus-test only the K8sHubble tests. If that + smoke test pass, we should be good to go.

**Comment by joestringer** [comment]
test-focus K8sHubbleTest

**Comment by joestringer** [comment]
Hit known flake https://github.com/cilium/cilium/issues/11955 in Travis, ignoring.

**Pull request merged**